src/Microsoft.Health.Dicom.Core/Messages/WorkItem/RetrieveWorkitemResponse.cs

```diff
@@ -0,0 +1,25 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using EnsureThat;
+using FellowOakDicom;
+
+namespace Microsoft.Health.Dicom.Core.Messages.Workitem;
+
+public sealed class RetrieveWorkitemResponse
+{
+    public RetrieveWorkitemResponse(WorkitemResponseStatus status, DicomDataset responseDataset, string message = null)
+    {
+        Dataset = EnsureArg.IsNotNull(responseDataset, nameof(responseDataset));
```

Comment:
Isn't dataset expected to be null when the Ups Instance is not found? Or do we not go through this path on failure?

Reply:
If it is a failure, it will be new DicomDataset()

Reply:
Makes sense. Thank you!